### PR TITLE
feat: Add stale workflow

### PR DESCRIPTION
**Pull request by siriak**
Add stale workflow that'll close old issues/PRs

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C/pull/860"></a>

### CODE REVIEW

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  schedule:
+    - cron: '0 0 * * *'
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v4
+        with:
+          stale-issue-message: 'This issue has been automatically marked as abandoned because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.'
+          close-issue-message: 'Please ping one of the maintainers once you add more information and updates here. If this is not the case and you need some help, feel free to ask for help in our [Gitter](https://gitter.im/TheAlgorithms) channel. Thank you for your contributions!'
```
Comment (Panquesito7) [inline .github/workflows/stale.yml:12]:
```suggestion
          close-issue-message: 'Please ping one of the maintainers once you add more information and updates here. If this is not the case and you need some help, feel free to ask for help in our [Gitter](https://gitter.im/TheAlgorithms) channel or our [Discord server](https://discord.gg/c7MnfGFGa6). Thank you for your contributions!'
```

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+          stale-pr-message: 'This pull request has been automatically marked as abandoned because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.'
+          close-pr-message: 'Please ping one of the maintainers once you commit the changes requested or make improvements on the code. If this is not the case and you need some help, feel free to ask for help in our [Gitter](https://gitter.im/TheAlgorithms) channel. Thank you for your contributions!'
```
Comment (Panquesito7) [inline .github/workflows/stale.yml:14]:
```suggestion
          close-pr-message: 'Please ping one of the maintainers once you commit the changes requested or make improvements on the code. If this is not the case and you need some help, feel free to ask for help in our [Gitter](https://gitter.im/TheAlgorithms) channel or our [Discord server](https://discord.gg/c7MnfGFGa6). Thank you for your contributions!'
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

**Review by Panquesito7 — APPROVED** [review]
Thanks! 😄

**Comment by siriak** [comment]
Are we waiting on 1 more review? I cannot merge myself here

**Comment by Panquesito7** [comment]
> Are we waiting on 1 more review? I cannot merge myself here

Oops, I forgot to merge 🤦‍♂️ sorry. Merging now.

**Pull request merged**